src/DeviceInterface/InterfacePrototype.js

```diff
@@ -7,14 +7,17 @@ const {
     sendAndWaitForAnswer,
     formatDuckAddress
 } = require('../autofill-utils')
-const {forms} = require('../scanForInputs')
+const {getInputMainType, getInputSubtype} = require('../Form/matching')
 const {
     formatFullName
 } = require('../Form/formatters')
 const EmailAutofill = require('../UI/EmailAutofill')
 const DataAutofill = require('../UI/DataAutofill')
+const {getInputConfigFromType} = require('../Form/inputTypeConfig')
 
 let attempts = 0
+let currentAttached
```

Comment:
is there any reason we're using more module-level variables here? My understanding is that their would only ever be a single instance of this class (via classes that extend it), so I'm curious we are not using class properties here? 

I'm aware that `attempts` was already there, and my question applies to that also

Reply:
Agreed, I will move them into the class.

I think the rationale perhaps might have been that this wasn't designed as a singleton pattern; which it isn't however other code ensures that there's only ever one instance of this class.